OSSM-4104: Service Mesh 2.5 Release Notes (2.5 Release) (#68434)
openshift/openshift-docs: modules/ossm-rn-new-features.adoc

[NOTE]
====
{DTProductName} (Tempo) Stack is not supported on {ibm-z-title}. 
====

=== OpenShift Service Mesh Console plugin

This release introduces a generally available version of the OpenShift {SMProductShortName} Console (OSSMC) plugin. 

The OSSMC plugin is an extension to the OpenShift Console that provides visibility into your Service Mesh. With the OSSMC plugin installed, a new Service Mesh menu option is available in the left-hand navigation of the web console, as well as new Service Mesh tabs that enhance existing Workloads and Service console pages. 

The features of the OSSMC plugin are very similar to those of the standalone Kiali Console. The OSSMC plugin does not replace the Kiali Console, and after installing the OSSMC plugin, you can still access the standalone Kiali Console. 

=== Istio OpenShift Routing (IOR) default setting change

> The default setting for Istio OpenShift Routing (IOR) has changed. Starting with this release, automatic routes are disabled by default for new instances of the `ServiceMeshControlPlane` resource.

Reviewer: Should we add a sentence for what this means for existing `ServiceMeshControlPlane` resources?

Reviewer: The word "new" hints that as an existing user, I shouldn't worry about this. I am concerned that people might miss this word and worry that it may cause IOR to disable for their existing mesh and for that to break their existing Gateways. Thus, I wonder if to be super clear, we should add a sentence like "This will not impact existing `ServiceMeshControlPlane` instances, which will continue to have IOR enabled by default."

Perhaps another question though - for engineering... if I update an existing `ServiceMeshControlPlane` instance, will the setting continue to default to enabled? (Just want to double check).

Author: Thanks for clarifying for me. I'll make an update.

Author: Done. Added sentence we vetted in slack.

This release adds improvements related to the following components and concepts.

> == New features {SMProductName} version 2.5

Reviewer 2: Per our docs guidelines each heading needs its own ID. I notice that few if any of the headings in this module have an ID. You don't have to add IDs for headings that are already merged in this module, but for each heading that you are adding in this PR, please add a unique ID:
```suggestion
[id="new-features-ossm-2-5"]
== New features {SMProductName} version 2.5
```

It's a pain, but without these IDs added manually to ensure uniqueness, the platform autogenerates them based on the text in the heading. If you ever have duplicate headings within an assembly/module, this creates duplicate IDs and I imagine that causes issues for our site.

> This release of {SMProductName} adds new features, addresses Common Vulnerabilities and Exposures (CVEs), contains bug fixes, and is supported on {product-title} 4.13 and later versions.

Reviewer 2: I think you can drop "versions" here, per the examples in the [ISG](https://www.ibm.com/docs/en/ibm-style?topic=references-product-versions#references-to-earlier-or-later-versions):
```suggestion
This release of {SMProductName} adds new features, addresses Common Vulnerabilities and Exposures (CVEs), contains bug fixes, and is supported on {product-title} 4.13 and later.
```

Author: Done

> This release ends maintenance support for OpenShift {SMProductShortName} version 2.2. If you are using OpenShift {SMProductShortName} version 2.2, you should upgrate to a supported version.

Author: done

> This release ends maintenance support for OpenShift {SMProductShortName} version 2.2. If you are using OpenShift {SMProductShortName} version 2.2, you should upgrate to a supported version.

Author: Done

=== Component versions for {SMProductName} version 2.5

|===
|Component |Version

|Istio
|1.18.5

|Envoy Proxy
|1.26.8

|Kiali
|1.73.4
|===

=== Istio 1.18 support

Service Mesh 2.5 is based on Istio 1.18, which brings in new features and product enhancements. While {SMProductName} supports many Istio 1.18 features, the following exceptions should be noted:

* Ambient mesh is not supported
* QuickAssist Technology (QAT) PrivateKeyProvider in Istio is not supported

=== Cluster-Wide mesh migration

This release adds documentation for migrating from a multitenant mesh to a cluster-wide mesh. For more information, see the following documentation:

> * About migrating to a cluster-wide mesh
> * Excluding namespaces from a cluster-wide mesh
> * Defining which namespaces receive sidecar injection in a cluster-wide mesh
> * Excluding individual pods from a cluster-wide mesh

Reviewer 2: [Per the ISG](https://www.ibm.com/docs/en/ibm-style?topic=references-citations#ibm-documentation), I think if you can't link to the doc, the title should be placed in quotation marks:
```suggestion
* "About migrating to a cluster-wide mesh"
* "Excluding namespaces from a cluster-wide mesh"
* "Defining which namespaces receive sidecar injection in a cluster-wide mesh"
* "Excluding individual pods from a cluster-wide mesh"
```

Author: Done

> === {SMProductName} operator on ARM-based clusters

Reviewer 2: ```suggestion
=== {SMProductName} Operator on ARM-based clusters
```

Author: Done

> This release provides the {SMProductName} Operator available on ARM-based clusters as a generally available feature. 

Reviewer 2: I think available can be dropped here:
```suggestion
This release provides the {SMProductName} Operator on ARM-based clusters as a generally available feature. 
```

=== Integration with {DTProductName} (Tempo) Stack

> This release introduces a generally available integration of the tracing extension provider(s). You can expose tracing data to the {DTProductName} (Tempo) stack by appending a named element and `zipkin` provider to the `spec.meshConfig.extensionProviders` specification. Then, a telemetry custom resource configures Istio proxies to collect trace spans and send them to the Tempo distributor service endpoint.

Reviewer 2: Question - is this term missing an article like "the" or "a" here?
```suggestion
This release introduces a generally available integration of the tracing extension provider(s). You can expose tracing data to the {DTProductName} (Tempo) stack by appending a named element and the `zipkin` provider to the `spec.meshConfig.extensionProviders` specification. Then, a telemetry custom resource configures Istio proxies to collect trace spans and send them to the Tempo distributor service endpoint.
```

Author: Done

[NOTE]
====
{DTProductName} (Tempo) Stack is not supported on {ibm-z-title}. 
====

=== OpenShift Service Mesh Console plugin

This release introduces a generally available version of the OpenShift {SMProductShortName} Console (OSSMC) plugin. 

The OSSMC plugin is an extension to the OpenShift Console that provides visibility into your Service Mesh. With the OSSMC plugin installed, a new Service Mesh menu option is available in the left-hand navigation of the web console, as well as new Service Mesh tabs that enhance existing Workloads and Service console pages. 

The features of the OSSMC plugin are very similar to those of the standalone Kiali Console. The OSSMC plugin does not replace the Kiali Console, and after installing the OSSMC plugin, you can still access the standalone Kiali Console. 

> 

Reviewer 2: These lines can be deleted

=== Istio OpenShift Routing (IOR) default setting change

The default setting for Istio OpenShift Routing (IOR) has changed. Starting with this release, automatic routes are disabled by default for new instances of the `ServiceMeshControlPlane` resource. 

For new  instances of the `ServiceMeshControlPlane` resources, you can use automatic routes by setting the `enabled` field to `true` in the `gateways.openshiftRoute` specification of the `ServiceMeshControlPlane` resource.

> [source,yaml]

Reviewer 2: Up to you but it might be helpful to give this a title:
```suggestion
.Example ServiceMeshControlPlane resource
[source,yaml]
```

Author: Done

----
apiVersion: maistra.io/v2
kind: ServiceMeshControlPlane
spec:
  gateways:
    openshiftRoute:
      enabled: true
----

> When upgrading existing instances of the `ServiceMeshControlPlane` resource to {SMProductName} version 2.5, automatic routes remain enabled by default.

Reviewer 2: Update is the preferred term:
```suggestion
When updating existing instances of the `ServiceMeshControlPlane` resource to {SMProductName} version 2.5, automatic routes remain enabled by default.
```

Author: Done

=== Istio proxy concurrency configuration enhancement

The `concurrency` parameter in the `networking.istio` API configures how many worker threads the Istio proxy runs. 

> For consistency across deployments, Istio now configures the `concurrency` parameter based upon the CPU limit allocated to the proxy container. For example, a limit of 2500m would set the `concurrency` parameter to 3. If you set the `concurrency` parameter to a different value, then Istio uses that value to configure how many threads the proxy runs instead of using the CPU limit. 

Reviewer 2: Since this is (I imagine) a value that you would see either in a file or CLI, it should be in literals:
```suggestion
For consistency across deployments, Istio now configures the `concurrency` parameter based upon the CPU limit allocated to the proxy container. For example, a limit of 2500m would set the `concurrency` parameter to `3`. If you set the `concurrency` parameter to a different value, then Istio uses that value to configure how many threads the proxy runs instead of using the CPU limit. 
```

Author: Done

> Previously, the default setting for the parameter was 2. 

Reviewer 2: Same comment as above:
```suggestion
Previously, the default setting for the parameter was `2`. 
```

Author: Done

=== Gateway API CRD versions
:FeatureName: {product-title} Gateway API support
include::snippets/technology-preview.adoc[]

> A new version of the Gateway API CRD is now available. Refer to the following table to determine which Gateway API version should be installed with the OpenShift {SMProductShortName} version you are using:

Reviewer 2: Does CRD mean custom resource definition? If so, I think that it should be spelled out in the first instance here:
```suggestion
A new version of the Gateway API custom resource definition (CRD) is now available. Refer to the following table to determine which Gateway API version should be installed with the OpenShift {SMProductShortName} version you are using:
```

Author: Done

|===
|Service Mesh Version | Istio Version | Gateway API Version | Notes

|2.5.x
|1.18.x
|0.6.2

> |Use experimental branch because `ReferenceGrand` is missing in v0.6.2

Reviewer 2: Suggestion:
```suggestion
|Use the experimental branch because `ReferenceGrand` is missing in v0.6.2
```

Author: Done

|2.4.x
|1.16.x
|0.5.1

> |For multitenant mesh deployment, all Gateway API CRDs must be present. Use experimental branch.

Reviewer 2: Suggestion:
```suggestion
|For multitenant mesh deployment, all Gateway API CRDs must be present. Use the experimental branch.
```

Author: Done

|===

== New features {SMProductName} version 2.4.6

> This release of {SMProductName} addresses Common Vulnerabilities and Exposures (CVEs), contains bug fixes, and is supported on {product-title} 4.12 and later versions.

Reviewer 2: I think you can drop "versions" here, per the examples in the [ISG](https://www.ibm.com/docs/en/ibm-style?topic=references-product-versions#references-to-earlier-or-later-versions): 
```suggestion
This release of {SMProductName} addresses Common Vulnerabilities and Exposures (CVEs), contains bug fixes, and is supported on {product-title} 4.12 and later.
```

Author: Done

* Ambient mesh is not supported

=== Cluster-Wide mesh migration

This release adds documentation for migrating from a multitenant mesh to a cluster-wide mesh. For more information, see the following documentation:

* About migrating to a cluster-wide mesh
* Excluding namespaces from a cluster-wide mesh
* Defining which namespaces receive sidecar injection in a cluster-wide mesh
* Excluding individual pods from a cluster-wide mesh

=== {SMProductName} operator on ARM-based clusters
This release provides the {SMProductName} Operator available on ARM-based clusters as a generally available feature. Images are available for Istio, Envoy, Prometheus, Kiali, Jaeger, and Grafana. 

=== Integration with {DTProductName} (Tempo) Stack

This release introduces a generally available integration of the tracing extension provider(s). You can expose tracing data to the {DTProductName} (Tempo) stack by appending a named element and `zipkin` provider to the `spec.meshConfig.extensionProviders` specification. Then, a telemetry custom resource configures Istio proxies to collect trace spans and send them to the Tempo distributor service endpoint.

> The following example configures the `ServiceMeshControlPlane` resource to connect to the Tempo Stack instance in the `tracing-system` namespace:

Reviewer 3: ```suggestion
The following example configures the `ServiceMeshControlPlane` resource to define an extensionProvider `zipkin` for sending tracing spans to a Tempo Stack. A Tempo Stack instance can be created in the `tracing-system` namespace after creating the `ServiceMeshControlPlane` and the `ServiceMeshMemberRoll` resources.
```

Author: done

[source,yaml]
----
kind: ServiceMeshControlPlane
apiVersion: maistra.io/v2
metadata:
  name: basic
  namespace: istio-system
spec:
...
  meshConfig:
    extensionProviders:
      - name: tempo
        zipkin:
          service: tempo-sample-distributor.tracing-system.svc.cluster.local
          port: 9411
  tracing:
    sampling: 10000
    type: None <1>
  version: v2.5
---
kind: ServiceMeshMemberRoll
apiVersion: maistra.io/v1
metadata:
  name: default
  namespace: istio-system
spec:
  members:
    - tracing-system
----
<1> The `spec.tracing.type` setting defines a Distributed Tracing instance and 2 Jaeger instances. Set `spec.tracing.type` to `None` when connecting to a Tempo Stack using an `extensionProvider` setting.

Configure the Kiali resource specification to enable a Kiali workload Traces dashboard. You can use the dashboard to view tracing query results.

[source,yaml]
----
apiVersion: kiali.io/v1alpha1
kind: Kiali
...
spec:  
  external_services:
    tracing:
      enabled: true
      in_cluster_url: 'http://tempo-sample-query-frontend.tracing-system.svc.cluster.local:16686'
      url: '[Tempo query frontend Route url]'
----

> 

Reviewer 3: ```suggestion
Create a Tempo Stack instance using the `Red Hat Tempo Operator` in the `tracing-system` namespace. See reference in :
https://docs.openshift.com/container-platform/4.14/distr_tracing/distr_tracing_tempo/distr-tracing-tempo-installing.html
 
```

Author: done

The following example configures the `ServiceMeshControlPlane` resource to define a `zipkin` extension provider for sending tracing spans to a Tempo Stack. A Tempo Stack instance can be created in the `tracing-system` namespace after creating the `ServiceMeshControlPlane` and the `ServiceMeshMemberRoll` resources.

[source,yaml]
----
kind: ServiceMeshControlPlane
apiVersion: maistra.io/v2
metadata:
  name: basic
  namespace: istio-system
spec:
...
  meshConfig:
    extensionProviders:
      - name: tempo
        zipkin:
          service: tempo-sample-distributor.tracing-system.svc.cluster.local
          port: 9411
  tracing:
    sampling: 10000
    type: None <1>
  version: v2.5
---
kind: ServiceMeshMemberRoll
apiVersion: maistra.io/v1
metadata:
  name: default
  namespace: istio-system
spec:
  members:
    - tracing-system
----

> <1> The `spec.tracing.type` setting defines a Distributed Tracing instance and 2 Jaeger instances. Set `spec.tracing.type` to `None` when connecting to a Tempo Stack using an `extensionProvider` setting.

Author: This procedure was moved to core docs.

|2.4.x
|1.16.x
|0.5.1
|For multitenant mesh deployment, all Gateway API CRDs must be present. Use experimental branch.
|===

=== Integration with {DTProductName} (Tempo) Stack
:FeatureName: {product-title} {DTProductName} (Tempo) stack
include::snippets/technology-preview.adoc[]

> This release introduces a technology preview integration of the tracing extension provider(s). You can expose tracing data to the {DTProductName} (Tempo) stack by appending a named element and `zipkin` provider to the `spec.meshConfig.extensionProviders` specification. Then, a telemetry custom resource configures Istio proxies to collect trace spans and send them to the Tempo distributor service endpoint.

Reviewer 3: There is a discussion about GA this integration in 2.5 release note. We may replace or remove the "technology preview" phrase this line.

Author: Done. It's now listed as GA.

> The following example configures the `ServiceMeshControlPlane` resource to create a Tempo Stack instance in the `tracing-system` namespace:

Reviewer 3: This line is not correct. SMCP cannot create a Tempo Stack. 
I would suggest the following:

After a Tempo Stack instance has been created in a mesh namespace "tracing-system", the following example configures a ServiceMeshControlPlane resource to connect to the Tempo Stack instance in the `tracing-system` namespace

Author: done

[source,yaml]
----
kind: ServiceMeshControlPlane
apiVersion: maistra.io/v2
metadata:
  name: basic
  namespace: istio-system
spec:
...
  meshConfig:
    extensionProviders:
      - name: tempo
        zipkin:

>           service: tempo-sample-distributor.tracing-system.svc.cluster.local <1>

Reviewer 3: This <1> is not correct. <1> should be marked in line 128  .spec.tracing.type

Author: done

          port: 9411
  tracing:
    sampling: 10000

>     type: None <2>

Reviewer 3: <1> <2> should all refer this line. You may combine them together.

Author: done

====

=== Component versions included in {SMProductName} version 2.5.0

|===
|Component |Version

|Istio
|1.18

|Envoy Proxy
|1.24.12

|Jaeger
|1.47.0

|Kiali
|1.73.11
|===

=== OpenShift Service Mesh Console plugin

This release introduces a generally available version of the OpenShift {SMProductShortName} Console (OSSMC) plugin. 

The OSSMC plugin is an extension to the OpenShift Console that provides visibility into your Service Mesh. With the OSSMC plugin installed, a new Service Mesh menu option on the left-hand navigation of the web console, as well as new Service Mesh tabs that enhance existing Workloads and Service console pages. 

The features of the OSSMC plugin are very similar to those of the standalone Kiali Console. The OSSMC plugin does not replace the Kiali Console, and after installing the OSSMC plugin, you can still access the standalone Kiali Console. 

> === Univeral Base Image 9

Reviewer 4: *Universal Base Image 9

Author: I see what you did you there, mi amigo! The fix has been submitted.

The features of the OSSMC plugin are very similar to those of the standalone Kiali Console. The OSSMC plugin does not replace the Kiali Console, and after installing the OSSMC plugin, you can still access the standalone Kiali Console. 

=== Cluster-Wide mesh migration

This release adds documentation for migrating from a multitenant mesh to a cluster-wide mesh. For more information, see the following documentation:

* About migrating to a cluster-wide mesh
* Excluding namespaces from a cluster-wide mesh
* Defining which namespaces receive sidecar injection in a cluster-wide mesh
* Excluding individual pods from a cluster-wide mesh

=== Gateway API CRD versions

A new version of the Gateway API CRD is now available. Refer to the following table to determine which Gateway API version should be installed with the OpenShift {SMProductShortName} version you are using.

|===
|Service Mesh Version | Istio Version | Gateway API Version | Notes

|2.5.x
|1.18.x
|0.6.2
|Use experimental branch because `ReferenceGrand` is missing in v0.6.2

|2.4.x
|1.16.x
|0.5.1
|For multitenant mesh deployment, all Gateway API CRDs must be present. Use experimental branch.
|===

> 

Author: @yxun Please review the ARM Support entry and provide feedback. The entry was created using information from when ARM Support was introduced as Technology Preview and updated for GA. Thanks!

Author: Addressed in slack

* {SMProductShortName} on ARM64 architecture is not supported.
* OpenTelemetry API remains a Technology Preview feature.

== New features {SMProductName} version 2.3.10
//Update with 2.5

> This release of {SMProductName} addresses Common Vulnerabilities and Exposures (CVEs), contains bug fixes, and is supported on {product-title} 4.13 and later versions.

Reviewer 2: Same comment as above about being able to drop versions:
```suggestion
This release of {SMProductName} addresses Common Vulnerabilities and Exposures (CVEs), contains bug fixes, and is supported on {product-title} 4.13 and later.
```
